### PR TITLE
Updated `documentquery` with operators like `$in`

### PR DESCRIPTION
**Pull request by BykhovDenis**

### CODE REVIEW

File: packages/core/src/storage.ts [hunk 1]
```diff
@@ -13,10 +13,23 @@
...
 //
 
-import type { Class, Doc, Ref } from './classes'
+import type { Class, Doc, Obj, Ref } from './classes'
 import type { Tx } from './tx'
 
-export type DocumentQuery<T extends Doc> = Partial<T>
+// eslint-disable-next-line @typescript-eslint/consistent-type-definitions
+export type QuerySelector<T> = {
+  $gt?: T
```
Comment (haiodo) [inline packages/core/src/storage.ts:21]: Why we need any other then $in?

Comment (BykhovDenis) [reply packages/core/src/storage.ts:21]: I don't know) Some overengineering, because I think `operators like $in` is more then only `$in`)

Comment (BykhovDenis) [reply packages/core/src/storage.ts:21]: I can delete it, and add when we necessary

File: packages/core/src/memdb.ts [hunk 2]
```diff
@@ -69,9 +69,16 @@ class MemDb {
     if (Object.prototype.hasOwnProperty.call(query, '_id')) {
       if (query._id === undefined) {
         result = []
-      } else {
+      } else if (typeof query._id === 'string') {
         const obj = this.objectById.get(query._id)
         result = obj !== undefined ? [obj] : []
+      } else {
```
Comment (haiodo) [inline packages/core/src/memdb.ts:75]: This should be as separate PR.
